public/app/plugins/datasource/influxdb/response_parser.ts

```diff
@@ -84,6 +85,72 @@ export default class ResponseParser {
 
     return table;
   }
+
+  async transformAnnotationResponse(options: any, data: any, target: InfluxQuery): Promise<AnnotationEvent[]> {
+    const rsp = toDataQueryResponse(data, [target] as DataQuery[]);
+
+    if (rsp) {
+      const table = this.getTable(rsp.data, target, {});
+      const list: any[] = [];
+      let titleCol: any = null;
+      let timeCol: any = null;
+      let timeEndCol: any = null;
+      const tagsCol: any = [];
+      let textCol: any = null;
+
+      each(table.columns, (column, index) => {
+        if (column.text.toLowerCase() === 'time') {
+          timeCol = index;
+          return;
+        }
+        if (column.text === options.annotation.titleColumn) {
+          titleCol = index;
+          return;
+        }
+        if (includes((options.annotation.tagsColumn || '').replace(' ', '').split(','), column.text)) {
+          tagsCol.push(index);
+          return;
+        }
+        if (column.text === options.annotation.textColumn) {
+          textCol = index;
+          return;
+        }
+        if (column.text === options.annotation.timeEndColumn) {
+          timeEndCol = index;
+          return;
+        }
+        // legacy case
+        if (!titleCol && textCol !== index) {
+          titleCol = index;
+        }
+      });
+
+      each(table.rows, (value) => {
+        const data = {
+          annotation: options.annotation,
+          time: +new Date(value[timeCol]),
+          title: value[titleCol],
+          timeEnd: value[timeEndCol],
+          // Remove empty values, then split in different tags for comma separated values
+          tags: flatten(
```

Comment:
Field mapping doesn't seems to work correctly after migration.  Example For the query `SELECT time,host,message,path from logs where time >= now() - 30m and time <= now()`,  the tags are not working as expected after migration.

### Query:
<img width="862" alt="image" src="https://user-images.githubusercontent.com/153843/158387945-6cca489d-08cd-4448-8aac-faf319566fc1.png">

### Before migration:
<img width="470" alt="image" src="https://user-images.githubusercontent.com/153843/158388231-241f55d4-4f70-4d5a-9e97-4ed2e19eeb63.png">

### After migration:

<img width="280" alt="image" src="https://user-images.githubusercontent.com/153843/158388429-4f0c7275-73fe-4aed-b8c9-716a72d6a72d.png">

Reply:
Also there are chances that user have multiple fields as tags.

<img width="762" alt="image" src="https://user-images.githubusercontent.com/153843/158389289-255dc959-82d9-4df1-ab01-465e0f3a5397.png">
<img width="368" alt="image" src="https://user-images.githubusercontent.com/153843/158389334-b7c04986-8490-4ccc-ab46-721149bced69.png">

Reply:
<img width="454" alt="Screenshot 2022-03-17 at 12 06 04" src="https://user-images.githubusercontent.com/90795735/158805085-4a91ea18-581c-4391-acdd-eb2bb491fb4e.png">